### PR TITLE
Add formats for String and Svg modules

### LINKED ISSUE #3: Expose class strings in a separate module

### LINKED ISSUE #14: Svg support
Styling Svg with TailwindCSS is pretty easy, however `Html.Attributes.class` silently fail when applied to Svg elements (because it relies on attributes rather than properties under the hood).  Could we create another module for Svg using the `Svg` classes?

### PR DESCRIPTION
**Pull request by sparksp**
This adds a `formats` configuration to allow String and Svg modules to be created.  The Svg module returns `Svg.Attributes` instead of `Html.Attributes`.

```js
module.exports = {
    plugins: [
        require("tailwindcss")('tailwind.config.js'),
        require("postcss-elm-tailwind")(
            { elmFile: "src/Html/Tailwind.elm"
            , elmModuleName: "Html.Tailwind"
            , formats:
                { string:
                    { elmFile : "src/String/Tailwind.elm"
                    , elmModuleName: "String.Tailwind"
                    }
                , svg:
                    { elmFile : "src/Svg/Tailwind.elm"
                    , elmModuleName: "Svg.Tailwind"
                    }
                }
            }
        ),
        require("autoprefixer")
    ]
};
```

Resolves #3, resolves #14.

### CODE REVIEW

**Comment by monty5811** [comment]
This is wonderful - thank you!
I'll review it and get it merged this week.

**Comment by monty5811** [comment]
I changed things around a bit - figured it was a good time to handle issue #3 too.

And this way we only have to parse the class names once and then generate as many outputs as we like.

Let me know what you think 👍

File: index.js [hunk 1]
```diff
@@ -19,35 +20,64 @@ module.exports = postcss.plugin("postcss-elm-tailwind", opts => {
...
-
   let cls, elm;
 
   cls = h.fixClass(selector);
   elm = h.toElmName(cls, opts);
 
-  classes.set(cls, h.elmFunction(cls, elm, htmlOrSvg));
-  elm_fns.push(elm);
+  classes.set(cls, elm);
+  elmFns.add(elm);
```
[the remark this answers is not in the document]
Comment (monty5811) [reply index.js:82]: Yeah, probably, just a left over right now

File: README.md [hunk 2]
```diff
@@ -80,6 +80,52 @@ module.exports = {
...
+
+```js
+module.exports = {
+  plugins: [
+    require("tailwindcss"),
+    require("postcss-elm-tailwind")({
+      elmFile: "src/Tailwind.elm",
+      elmModule: "Tailwind",
+      formats: {
+        svg: {
```
[the remark this answers is not in the document]
Comment (monty5811) [reply README.md:97]: :grinning:

File: test/tests.js [hunk 3]
```diff
@@ -26,16 +26,7 @@ describe("cleanOpts", () => {
       elmModuleName: "TW",
       prefix: "tw--",
       nameStyle: "snake",
-      svg: false
-    });
-  });
-  it("should ensure svg is a boolean", () => {
```
[the remark this answers is not in the document]
Comment (monty5811) [reply test/tests.js]: I think, just leave it, if it is anything other than an object, then the plugin will crash :man_shrugging:

File: index.js [hunk 4]
```diff
@@ -19,35 +20,64 @@ module.exports = postcss.plugin("postcss-elm-tailwind", opts => {
...
-    });
+    if (opts.formats.string) {
+      writeStringFile(opts.formats.string, elmFns, classes);
+    }
   };
 });
 
+function writeMainFile({ elmFile, elmModuleName }, elmFns, classes) {
+  const elmModule = h.elmHeaderHtml(elmModuleName, elmFns) +
+    h.elmBody({ type: "Html.Attribute msg", fn: "A.class " }, classes);
```
[the remark this answers is not in the document]
Comment (monty5811) [reply index.js:37]: :+1:  I'm not fussed, what you have said is probably nicer

**Comment by monty5811** [comment]
Feel free to make the changes in line with your comments.
I'm happy to, but I probably won't get to it for a couple of days.

Also, add yourself in [here](https://github.com/monty5811/postcss-elm-tailwind/blob/master/CONTRIBUTORS.md) if you get to it before I do :smile:

**Comment by sparksp** [comment]
I've resolved all the feedback but was getting intermittent crashes with svg but no string in my config, so I cleaned up the options a bit more, realised we could use splats (`...`) to simplify the whole cleanOpts method, then what I had was pretty easy to async too so I finished off that while I was at it.  Hopefully this is all acceptable!?

**Comment by sparksp** [comment]
Here's where I got to with the async promise work... https://github.com/sparksp/postcss-elm-tailwind/pull/1

I don't know why, but when I linked it locally it worked perfectly, but when I tried to use it direct from GitHub then tailwind silently stopped building.

**Pull request merged**